S2VX.Game/Editor/ToolState/SelectToolState.cs

```diff
@@ -35,26 +36,93 @@ public class SelectToolState : S2VXToolState {
             return mouseInXRange && mouseInYRange;
         }
 
+        // https://gamedev.stackexchange.com/a/86784
+        private static Vector2 GetRotatedPoint(Vector2 point, Vector2 center, float rotation) {
+            var translatedX = point.X - center.X;
+            var translatedY = point.Y - center.Y;
+
+            var rotatedX = translatedX * (float)Math.Cos(rotation) - translatedY * (float)Math.Sin(rotation);
+            var rotatedY = translatedX * (float)Math.Sin(rotation) + translatedY * (float)Math.Cos(rotation);
+
+            return new Vector2(rotatedX + center.X, rotatedY + center.Y);
+        }
+
+        private bool MouseIsOnNote(Vector2 mousePos, Note note) {
```

Comment:
I prefer naming bool functions with Is at the beginning so you can more easily identify them (same with "MouseIsOnTimelineNote")

Reply:
Agreed. In general, there is a convention for boolean functions to be phrased as a yes or no question: https://stackoverflow.com/questions/1370840/naming-conventions-what-to-name-a-method-that-returns-a-boolean